script.js

```diff
@@ -1 +1,402 @@
+
+  // match message will disappear after a set time
+  setTimeout(() => {
+    boardElement.removeChild(matchMessageContainer);
+
+    canClick = true;
+
+    // tell player to click on another square
+    output('It is a match! <br> Click on another square to continue!');
+  }, 3000);
+};
+
+// flip cards and check for pairs when a square is clicked
+const squareClick = (squareClickElement, column, row, boardElement) => {
+  console.log(squareClickElement);
+  console.log('FIRST CARD', firstCard);
+  console.log('CLICKED CARD', board[column][row]);
+
+  if (firstCard === null) {
+    firstCard = board[column][row];
+    firstClickedSquare = squareClickElement;
+
+    // turn this card over
+    firstClickedSquare.innerText = `${firstCard.display},${firstCard.suitSymbol}`;
+
+    // let player know to click another card
+    output('Click on another square to flip over the 2nd card!');
+  } else if (firstClickedSquare === squareClickElement) {
+    output('Clicked on same card. Please click on another card');
+    // do nothing if player clicked on same card
+  }
+  else if (board[column][row].name === firstCard.name
+  && board[column][row].suit === firstCard.suit) {
+    secondClickedSquare = squareClickElement;
+
+    // turn this card over
+    secondClickedSquare.innerText = `${board[column][row].display},${board[column][row].suitSymbol}`;
+    console.log('match');
+
+    // increment number of matches by 1
+    numOfMatches += 1;
+
+    // display winning message and end game if player matches everything
+    if (numOfMatches === matchesNeededToWin) {
+      gameStatus = 'stopGame';
+
+      const specialMessage = document.createElement('img');
+      specialMessage.classList.add('special-message');
+      specialMessage.src = 'https://www.birthdaywishes.expert/wp-content/uploads/2018/06/Success.jpg';
+      document.body.appendChild(specialMessage);
+
+      setTimeout(() => {
+        document.body.removeChild(specialMessage);
+        output(`Congratulations ${playerName}. You have matched all cards! Please refresh to play again`);
+      }, 5000);
+
+      clearInterval(timerId);
+    }
+
+    // let player know that it is a match and to pick another card
+    // and clear output meesage
+    output('');
+    createMatchMessage(boardElement);
+
+    // empty first card
+    firstCard = null;
+  } else {
+    // empty first card
+    firstCard = null;
+
+    secondClickedSquare = squareClickElement;
+
+    // turn this card over
+    secondClickedSquare.innerText = `${board[column][row].display},${board[column][row].suitSymbol}`;
+
+    // let player know that it is not a match and to pick another card
+    output('Oops your cards did not match! <br> Click on another square to try again!');
+
+    // turn this card and first card back over
+    flipOverSquares(firstClickedSquare, secondClickedSquare);
+  }
+
+  scoreMessage.innerText = `Number of Matches: ${numOfMatches}`;
+};
+
+// make deck
+const makeDeck = () => {
```

Comment:
Great idea moving these helper functions to just above where they are used for better grouping of functions based on context!